packages/sitecore-jss/src/utils/resolve-url.ts

```diff
@@ -0,0 +1,50 @@
+import isServer from './is-server';
+import querystring from 'querystring';
+
+/**
+ * note: encodeURIComponent is available via browser (window) or natively in node.js
+ * if you use another js engine for server-side rendering you may not have native encodeURIComponent
+ * and would then need to install a package for that functionality
+ * @param {Object} params
+ */
+function getQueryString(params: querystring.ParsedUrlQueryInput) {
+  return Object.keys(params)
+    .map((k) => `${encodeURIComponent(k)}=${encodeURIComponent(String(params[k]))}`)
+    .join('&');
+}
+
+/**
+ * Resolves a base URL that may contain query string parameters and an additional set of query
+ * string parameters into a unified string representation.
+ * @param {string} urlBase the base URL that may contain query string parameters
+ * @param {Object} params query string parameters
+ * @returns a URL string
+ * @throws {RangeError} if the provided url is an empty string
+ */
+function resolveUrl(urlBase: string, params: querystring.ParsedUrlQueryInput = {}): string {
```

Comment:
`resolveUrl` should prob have some unit test coverage.

Reply:
done